src/orders/controller/orders.controller.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import OrdersMapper from '../mapper/orders.mapper';
+import { OrdersService } from '../service/orders.service';
+import Order from '../domain/order.domain';
+import { CustomersService } from 'src/customers/service/customers.service';
+import OrderDTO from '../dto/order.dto';
+import { ApiResponse } from '@nestjs/swagger';
+
+@Controller('orders')
+export class OrdersController {
+  constructor(
+    private ordersMapper: OrdersMapper,
+    private ordersService: OrdersService,
+    private customersService: CustomersService,
+  ) {}
+
+  @ApiResponse({ status: 200, description: 'Returns the orders' })
+  @Get()
+  async getOrders(): Promise<Order[]> {
+    return await this.ordersService.getOrders();
+  }
+
+  @ApiResponse({ status: 200, description: 'Returns the order' })
+  @ApiResponse({ status: 404, description: "The order can't be found" })
+  @Get(':id')
+  async getOrderById(@Param() { id }: { id: string }): Promise<Order | null> {
+    const order = await this.ordersService.getOrderById(id);
+    if (!order) {
+      throw new NotFoundException();
+    }
+    return order;
+  }
+
+  @ApiResponse({
+    status: 201,
+    description: 'The order was created successfully',
+  })
+  @Post()
+  async createOrder(@Body() orderDTO: OrderDTO): Promise<Order> {
```

Comment:
Something is missing from this flow: can we always create an order? Does creating an order without products make sense? 😄  Don't we need to check something beforehand?

- We need to send as input the "order details"
- We need to verify whether the products exist and the required quantity is in stock 
- We need to update the stock (decrease it).

Reply:
### TODO!!!
This will be revised on Monday